cmd/world/cardinal/dev.go

```diff
@@ -108,18 +109,32 @@ func runRedis() error {
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
 
-	err := cmd.Start()
-	if err != nil {
-		fmt.Println("Failed to start Redis container. Retrying after cleanup...")
-		cleanupErr := cleanup()
-		if cleanupErr != nil {
-			return err
-		}
-
-		err := sh.Run("docker", "run", "-d", "-p", fmt.Sprintf("%s:%s", RedisPort, RedisPort), "--name", "cardinal-dev-redis", "redis")
+	handlerError := func(err error) error {
 		if err != nil {
-			return err
+			fmt.Println("Failed to start Redis container. Retrying after cleanup...")
+			cleanupErr := cleanup()
+			if cleanupErr != nil {
+				return err
+			}
+
+			err := sh.Run("docker", "run", "-d", "-p", fmt.Sprintf("%s:%s", RedisPort, RedisPort), "--name", "cardinal-dev-redis", "redis")
+			if err != nil {
+				return err
+			}
+			return nil
 		}
+		return nil
+	}
+
+	err := cmd.Start()
```

Comment:
cmd.Run() calls cmd.Start and cmd.Wait for you, so I think this can be a 1 line PR.

To make sure redis is ready, you could add some code on line 46 (Between runRedis and run Cardinal) to make sure you can connect to redis. something like `docker exec starter-game-redis-1 redis-cli ping`